src/tests/ftest/erasurecode/rebuild_fio.py

```diff
@@ -101,6 +101,9 @@ def execution(self, rebuild_mode):
             # Read and verify the original data.
             self.fio_cmd.run()
 
+        # Pre-teardown: make sure rebuild is done before too-quickly trying to destroy container.
+        self.pool.wait_for_rebuild_to_end()
```

Comment:
Does this need to happen whether the test passes or fails? Or only if we get this far without errors?

Reply:
I think it needs to happen if a rebuild has been triggered, so if the execution has gotten to any of the stop_ranks calls.

Just quickly (admittedly maybe too quickly) looking at the preceding code execution flow,  if the code gets here it must have successfully performed the preceding steps? Since if they failed they would have either raised an exception not caught here (e.g., raised from some of the code in those ErasureCodeFio class abstractions) or directly invoked self.fail()?

Reply:
Right. If we get here, you can assume everything above was successful. But if something above failed - I.e. raised an exception - this `wait_for_rebuild_to_end()` would not run. Maybe that's okay? Just checking